models/Asset/Image/Thumbnail/Config.php

```diff
@@ -965,4 +963,19 @@ public function getAutoFormatThumbnailConfigs(): array
 
         return $autoFormatThumbnails;
     }
+
+    /**
+     * @internal
+     */
+    public function generateAutoName(): void
+    {
+        try {
+            $serialized = Serialize::serialize($this->getItems());
```

Comment:
Isn't that dangerous as it would generate new files again and again?

Reply:
you are right, this will generate auto thumbnails on every request. I'll fix it.

Reply:
fixed with `opis/closure` to serialize thumbnail closure items.

Reply:
Careful, as this library is not compatible with PHP 8.1. See note below.